File: app/src/main/java/net/kibotu/android/bloodhound/ContextExtensions.java
package net.kibotu.android.bloodhound;

import android.content.Context;
import android.support.annotation.NonNull;

/**
 * Created by Nyaruhodo on 07.05.2016.
 */
public class ContextExtensions {

    public static String getApplicationName(@NonNull final Context context) {
        int stringId = context.getApplicationInfo().labelRes;
        return context.getString(stringId);
    }

    public static String getPackageName(@NonNull final Context context) {
        return context.getApplicationInfo().packageName;
    }
}

File: lib/src/main/java/net.kibotu/bloodhound/ContextExtensions.kt
/*
 * Created by [Jan Rabe](https://about.me/janrabe)
 */

@file:JvmName("Extensions")

package net.kibotu.bloodhound

import android.annotation.TargetApi
import android.content.Context
import android.os.Build
import android.os.Bundle

internal val Context.ApplicationName: String
    get() = applicationInfo.loadLabel(packageManager).toString()

internal val Context.PackageName: String
    get() = applicationInfo.packageName

internal val Context.versionName: String
    get() = packageManager.getPackageInfo(packageName, 0).packageName

internal val Context.versionCode: Int
    get() = packageManager.getPackageInfo(packageName, 0).versionCode

internal val Context.longVersionCode: Long
    @TargetApi(Build.VERSION_CODES.P)
    get() = packageManager.getPackageInfo(packageName, 0).longVersionCode

internal fun Map<String, String>.toBundle(): Bundle? {
    val bundle = Bundle()
    forEach { (key, value) -> bundle.putString(key, value) }
    return bundle
}